src/classes/RD2_DataMigrationMapper.cls

```diff
@@ -68,7 +68,8 @@ public class RD2_DataMigrationMapper {
      * @return List<RecurringDonationSchedule__c> Schedule records to insert
      */
     public List<RecurringDonationSchedule__c> getScheduleRecords() {
-        return new List<RecurringDonationSchedule__c>();
+        RD2_ScheduleService scheduleService = new RD2_ScheduleService();
```

Comment:
The schedule record can be built only for the enhanced RD, so we need a check for `isConverted()` before returning schedule(s).
If the call is made to get schedules, but the record is not converted, let's generate an exception.

Reply:
@RadSFDeveloper, if the record is not converted, do you suggest to throw an custom exception here or error message?

Reply:
Custom exception: new `public class MigrationException extends Exception { }` defined in the `RD2_DataMigrationMapper`.

Reply:
I don't think the check is required here. This method will be called by the RD2_DataMigration_BATCH.handleConversion() method. In that method, it will only call this if the RD itself was converted. As a result, this method will never be called unless the RD was converted by the RD2_DataMigrationWrapper.

Reply:
@force2b and @RadSFDeveloper , Can you confirm if this check is not needed? I was about to make these changes.

Reply:
@force2b , I see it is handled in batch class, will not do these change then, thanks!

Reply:
@meena-sfdc , the check is not needed, but let's make changes to the test method.

Reply:
Good call out on the tests.